diff --git a/renderer/premake5.lua b/renderer/premake5.lua
index 57d5a28d..8ee9f5bf 100644
--- a/renderer/premake5.lua
+++ b/renderer/premake5.lua
@@ -6,11 +6,11 @@ dofile(RIVE_RUNTIME_DIR .. '/decoders/premake5_v2.lua')
 dofile(RIVE_RUNTIME_DIR .. '/dependencies/premake5_glfw_v2.lua')
 
 newoption({ trigger = 'with-skia', description = 'use skia' })
-if _OPTIONS['with-skia'] then
+if _OPTIONS['with-skia'] and not _OPTIONS['with-libs-only'] then
     dofile(RIVE_RUNTIME_DIR .. '/skia/renderer/premake5_v2.lua')
 end
 
-if not _OPTIONS['with-webgpu'] then
+if not _OPTIONS['with-webgpu'] and not _OPTIONS['with-libs-only'] then
     project('path_fiddle')
     do
         dependson('rive')
diff --git a/renderer/premake5_pls_renderer.lua b/renderer/premake5_pls_renderer.lua
index 266b3ed4..4bbc995f 100644
--- a/renderer/premake5_pls_renderer.lua
+++ b/renderer/premake5_pls_renderer.lua
@@ -70,6 +70,12 @@ end
 
 filter({})
 
+newoption({
+    trigger = 'with-libs-only',
+    description = 'only builds the libraries',
+})
+
+
 newoption({
     trigger = 'with_wagyu',
     description = 'compile in support for wagyu webgpu extensions',
