diff --git a/build/rive_build_config.lua b/build/rive_build_config.lua
--- a/build/rive_build_config.lua
+++ b/build/rive_build_config.lua
@@ -449,7 +449,7 @@ if _OPTIONS['for_android'] then
 
     -- Detect the NDK.
     local EXPECTED_NDK_VERSION = 'r27c'
-    local NDK_LONG_VERSION_STRING = "27.2.12479018"
+    local NDK_LONG_VERSION_STRING = "25.2.9519653"
     if _OPTIONS['for_unreal'] then
         EXPECTED_NDK_VERSION = '25.1.8937393'
         NDK_LONG_VERSION_STRING = '25.1.8937393'
diff --git a/premake5_v2.lua b/premake5_v2.lua
--- a/premake5_v2.lua
+++ b/premake5_v2.lua
@@ -43,6 +43,12 @@ do
 end
 filter({})
 
+filter({ 'options:with_pthread' })
+do
+    buildoptions({ '-pthread' })
+    linkoptions({ '-pthread' })
+end
+
 dependencies = path.getabsolute('dependencies/')
 dofile(path.join(dependencies, 'premake5_harfbuzz_v2.lua'))
 dofile(path.join(dependencies, 'premake5_sheenbidi_v2.lua'))
@@ -296,6 +302,12 @@ newoption({
     description = 'Compiles in RenderCanvas and Ore GPU abstraction layer.',
 })
 
+-- Optional pthread enabling (used e.g. for Emscripten wagyu builds)
+newoption({
+    trigger = 'with_pthread',
+    description = 'Enable pthread compile/link flags (-pthread)',
+})
+
 newoption({
     trigger = 'with_rive_docs',
     description = 'Indicates building for use with the docs generator.',
diff --git a/renderer/premake5.lua b/renderer/premake5.lua
--- a/renderer/premake5.lua
+++ b/renderer/premake5.lua
@@ -6,11 +6,11 @@ dofile(RIVE_RUNTIME_DIR .. '/decoders/premake5_v2.lua')
 dofile(RIVE_RUNTIME_DIR .. '/dependencies/premake5_glfw_v2.lua')
 
 newoption({ trigger = 'with-skia', description = 'use skia' })
-if _OPTIONS['with-skia'] then
+if _OPTIONS['with-skia'] and not _OPTIONS['with-libs-only'] then
     dofile(RIVE_RUNTIME_DIR .. '/skia/renderer/premake5_v2.lua')
 end
 
-if not _OPTIONS['with-webgpu'] then
+if not _OPTIONS['with-webgpu'] and not _OPTIONS['with-libs-only'] then
     project('path_fiddle')
     do
         dependson('rive')
@@ -186,7 +186,7 @@ if not _OPTIONS['with-webgpu'] then
     end
 end
 
-if _OPTIONS['with-webgpu'] or _OPTIONS['with-dawn'] then
+if (_OPTIONS['with-webgpu'] or _OPTIONS['with-dawn']) and not _OPTIONS['with-libs-only'] then
     project('webgpu_player')
     do
         kind('ConsoleApp')
diff --git a/renderer/premake5_pls_renderer.lua b/renderer/premake5_pls_renderer.lua
--- a/renderer/premake5_pls_renderer.lua
+++ b/renderer/premake5_pls_renderer.lua
@@ -180,6 +180,11 @@ do
     defines({ 'RIVE_WEBGPU=' .. _OPTIONS['webgpu-version'] })
 end
 
+newoption({
+    trigger = 'with-libs-only',
+    description = 'only builds the libraries',
+})
+
 filter({})
 
 newoption({
